src/main/scala/regmapper/Annotation.scala

```diff
@@ -2,22 +2,186 @@
+  def duplicate(n: ModuleName): RegFieldDescAnnotation = this.copy(n)
+}
+
+
+/** ********************************************************************************/
+
+case class RegFieldDescSer(
+  byteOffset: String,
+  bitOffset: Int,
+  bitWidth: Int,
+  name: String,
+  resetValue: BigInt,
+  accessType: String,
+  writeType: String,
+  rdAction: String,
+  desc: String,
+  group: String,
+  groupDesc: String,
+  volatile: Boolean = false,
+  hasReset: Boolean = false,
+  enumerations: Map[BigInt, (String, String)] = Map()
+)
+
+case class RegFieldSer(
+  regFieldName: String,
+  desc: RegFieldDescSer
+)
+
+case class RegistersSer(
+  displayName: String,
+  instanceCounter: Int,
+  baseAddress: BigInt,
+  regFields: Seq[RegFieldSer]
+)
+
+/**
+  * Firrtl annotation
+  *
+  * @param target
+  * @param regMappingSer
+  */
+case class RegFieldDescMappingAnnotation(
+  target: ModuleName,
+  regMappingSer: RegistersSer) extends SingleTargetAnnotation[ModuleName] {
+  def duplicate(n: ModuleName): RegFieldDescMappingAnnotation = this.copy(target = n)
+}
+
+/**
+  * Chisel Annotation
+  *
+  * This creates the firrtl annotation
+  *
+  * @param target
+  * @param regMappingSer
+  */
+case class RegMappingChiselAnnotation(
+  target: RawModule,
+  regMappingSer: RegistersSer) extends ChiselAnnotation with RunFirrtlTransform {
+  def toFirrtl: RegFieldDescMappingAnnotation = RegFieldDescMappingAnnotation(target.toNamed, regMappingSer)
+
+  def transformClass: Class[RegMappingDumpTransform] = classOf[RegMappingDumpTransform]
+}
+
+class RegMappingDumpTransform extends Transform {
+  def inputForm: CircuitForm = LowForm
+
+  def outputForm: CircuitForm = LowForm
+
+  def execute(state: CircuitState): CircuitState = {
+    state.annotations.foreach {
+      case RegFieldDescMappingAnnotation(t, desc) => println(desc)
     }
+    state
+  }
+}
+
+object GenRegDescsAnno {
+  // Each class which has regmaps has instance counters
+  private var instanceCounters = scala.collection.mutable.Map[String, Int]()
+
+  def addInstanceCounter(key: String) : Unit = {
+    instanceCounters +=  ( key -> 0)
+  }
+
+  def getInstanceCount(name: String, baseAddress: BigInt) : Int = {
+    val nameAddress = s"${name}.${baseAddress}"
+
+    if (! (instanceCounters isDefinedAt nameAddress)) {
+      addInstanceCounter(nameAddress)
+    }
+    val cnt = instanceCounters(nameAddress)
+    val newCnt = cnt + 1
+    instanceCounters(nameAddress) = newCnt
+    cnt
+  }
+
+  def makeRegMappingSer(rawModule: RawModule,
+    moduleName: String,
+    baseAddress: BigInt,
+    width: Int,
+    byteOffset: Int,
+    bitOffset: Int,
+    regField: RegField): RegFieldSer = {
+
+    val anonRegFieldName = s"unnamedRegField${byteOffset.toHexString}_${bitOffset}"
+    val regFieldName = regField.desc.map {_.name}.getOrElse("")
+    val selectedRegFieldName = if (regFieldName.isEmpty /* selectedName.isEmpty */) anonRegFieldName else regFieldName
```

Comment:
If you implement my change above you can delete this and just assign `name = regFieldName` below.